src/preferences/dialog/dlgpreferencepage.cpp

```diff
@@ -21,38 +21,30 @@ QUrl DlgPreferencePage::helpUrl() const {
     return QUrl();
 }
 
-void DlgPreferencePage::setScrollSafeGuardForAllInputWidgets(QObject* obj) {
+void DlgPreferencePage::setScrollSafeGuardForAllInputWidgets(QObject* pObj) {
     // Set the focus policy to for scrollable input widgets and connect them
-    // to the custom event filter
-    for (auto* ch : obj->children()) {
-        // children() does not descend into QGroupBox,
-        // so we need to do it manually
-        QGroupBox* gBox = qobject_cast<QGroupBox*>(ch);
-        if (gBox) {
-            setScrollSafeGuardForAllInputWidgets(gBox);
-            continue;
-        }
+    // to the custom event filter.
+    // Note: finding all relevant widgets with pObj->findchildren<Type*>
+    // is much faster than with
+    // for (auto* ch : pObj->children()) { qobject_cast<Type*<(ch); }
+    auto pCombos = pObj->findChildren<QComboBox*>();
+    for (auto* pCombo : pCombos) {
+        setScrollSafeGuard(pCombo);
+    }
```

Comment:
nit: add another function that shortens this to one function template call. A private template function would work. You can declare it in the header and then only implement it in the `.cpp` using 
```cpp
template<typename T>
void DlgPreferencePage::setScrollSafeGuardForChildrenOfType(QObject* pObj) {
    QList<T*> children = pObj->findChildren<T*>();
    for (T* pChild : children) {
        setScrollSafeGuard(pChild);
    }
}

Reply:
Perfect, thanks!
Took me a while to figure I need to call it like this
```cpp
setScrollSafeGuardForChildrenOfType<QComboBox>(pObj);
```

Reply:
yeah. sorry for not providing an example.

Reply:
Never mind, I didn't mean it like that actually. It's just been late and I didn't notice I'm using typename functions all the time, like here, qobject_cast.